### PR TITLE
Verify signature of SAML responses

### LINKED ISSUE #9: Verify signature of SAML responses
New settings:

* `PROXY_SERVICE_IDENTITY_PROVIDER['CERTIFICATE_FILE']` (string, optional) - a path to the file with Identity Provider's certificate used to sign its responses.

### PR DESCRIPTION
**Pull request by fenryxo**
Closes #9

### CODE REVIEW

File: eidas_node/connector/demo/views.py [hunk 1]
```diff
@@ -103,8 +103,15 @@ def post(self, request: HttpRequest) -> HttpResponse:
         """Handle a HTTP POST request."""
         saml_response_xml = b64decode(self.request.POST.get('SAMLResponse', '').encode('ascii')).decode('utf-8')
 
-        # Reformat with pretty printing
         if saml_response_xml:
+            # Verify signatures
+            cert_file = (CONNECTOR_SETTINGS.service_provider['response_signature'] or {}).get('cert_file')
+            if cert_file:
```
[the remark this answers is not in the document]
Comment (fenryxo) [reply eidas_node/connector/demo/views.py]: The default value of `StringSettings` is `''` not `None`. Should keep this condition as is or add `default=None, min_length=1` to all optional `StringSettings`?

File: eidas_node/tests/test_saml.py [hunk 2]
```diff
@@ -301,6 +302,28 @@ def test_str(self):
...
         self.assertEqual(str(SAMLResponse(None, None)), 'relay_state = None, document = None')
 
+    def test_assertion_none(self):
+        root = Element(Q_NAMES['saml2p:Response'])
+        self.assertIsNone(SAMLResponse(ElementTree(root)).assertion)
+
+    def test_assertion_exists(self):
+        root = Element(Q_NAMES['saml2p:Response'])
+        assertion = SubElement(root, Q_NAMES['saml2:Assertion'])
+        self.assertIs(SAMLResponse(ElementTree(root)).assertion, assertion)
```
[the remark this answers is not in the document]
Comment (fenryxo) [reply eidas_node/tests/test_saml.py]: There is a semantic difference. `assertIs` makes it clear that `SAMLResponse(ElementTree(root)).assertion` must return exactly the same object, not just an object which happens to be equal.

File: eidas_node/connector/demo/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tpazderka) [reply eidas_node/connector/demo/views.py]: I forgot about the `StringSetting`.

File: eidas_node/tests/test_saml.py [hunk 3]
```diff
@@ -301,6 +302,28 @@ def test_str(self):
...
+        encrypted_assertion = SubElement(root, Q_NAMES['saml2:EncryptedAssertion'])
+        decrypted_assertion = SubElement(encrypted_assertion, Q_NAMES['saml2:Assertion'])
+        self.assertIs(SAMLResponse(ElementTree(root)).assertion, decrypted_assertion)
+
+    def test_assertion_too_many(self):
+        root = Element(Q_NAMES['saml2p:Response'])
+        SubElement(root, Q_NAMES['saml2:Assertion'])
+        SubElement(SubElement(root, Q_NAMES['saml2:EncryptedAssertion']), Q_NAMES['saml2:Assertion'])
+        with self.assertRaisesMessage(ParseError, 'Too many assertion elements'):
+            _ = SAMLResponse(ElementTree(root)).assertion
```
[the remark this answers is not in the document]
Comment (fenryxo) [reply eidas_node/tests/test_saml.py]: Fixed.

**Comment by fenryxo** [comment]
Rebased.

**Pull request merged**